src/Evolution/Systems/CurvedScalarWave/Worldtube/KerrSchildDerivatives.cpp

```diff
@@ -0,0 +1,56 @@
+
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include "Evolution/Systems/CurvedScalarWave/Worldtube/KerrSchildDerivatives.hpp"
+
+#include <cstddef>
+
+#include "DataStructures/Tensor/EagerMath/Magnitude.hpp"
+#include "DataStructures/Tensor/EagerMath/Trace.hpp"
+#include "DataStructures/Tensor/Tensor.hpp"
+#include "Utilities/Gsl.hpp"
+
+namespace CurvedScalarWave::Worldtube {
+tnsr::iAA<double, 3> spatial_derivative_inverse_ks_metric(
+    const tnsr::I<double, 3>& pos) {
+  const double r = get(magnitude(pos));
+  const double one_over_r = 1. / r;
+  const double one_over_r_2 = square(one_over_r);
+  const double one_over_r_3 = one_over_r_2 * one_over_r;
+  const double one_over_r_4 = one_over_r_3 * one_over_r;
+  const double one_over_r_5 = one_over_r_4 * one_over_r;
+
+  tnsr::iAA<double, 3> di_imetric{};
+  tnsr::ii<double, 3> delta_ll{0.};
+  tnsr::Ij<double, 3> delta_ul{0.};
+  tnsr::i<double, 3> pos_lower{};
+
+  for (size_t i = 0; i < 3; ++i) {
+    delta_ll.get(i, i) = 1.;
+    delta_ul.get(i, i) = 1.;
+    pos_lower.get(i) = pos.get(i);
+  }
+
+  const auto d_imetric_ij = tenex::evaluate<ti::i, ti::J, ti::K>(
+      6. * pos(ti::J) * pos(ti::K) * pos_lower(ti::i) * one_over_r_5 -
+      2. * delta_ul(ti::J, ti::i) * pos(ti::K) * one_over_r_3 -
+      2. * delta_ul(ti::K, ti::i) * pos(ti::J) * one_over_r_3);
```

Comment:
[optional] Consider using a Horner-like evaluation here and elsewhere: instead of `a * (1/r^5) + b * (1/r^3)`, do `(a * (1/r^2) + b) * (1/r^3)`.